### ISSUE TITLE
Reduce set of power-down / power-up commands

### ISSUE THREAD

**Issue by Sv3n** [tukl-msd/DRAMPower#26]
We currently have separate meta-commands for fast/slow precharge/active power down, which is quite different from the simple PDN / PDX commands that a DRAM supports. An effort can be made to reduce the number of different power-down commands we support:
- A real DRAM selects the fast/slow mode at boot time. We can emulate this by adding a new boolean parameter to select fast or slow mode.
- We can (and do) keep the state of the memory (active or precharged) already anyway, so there is no need to encode it in the command. This gets rid of the _F and _S distinction.

To summarize, we can remove these power-down meta-commands:
    PDN_F_PRE
    PDN_S_PRE
    PDN_F_ACT
    PDN_S_ACT
    PUP_PRE
    PUP_ACT
- and add:
  PDN,
  PUP

And use the bank state to go into precharge or active power down.

The motivation behind this is to stick closer to the type of commands a dram controller actually issues, and reduce the command set.

**Comment by myzinsky** [comment]
I agree to get rid of the _F and _S distinciton.

But, to explore different PDN strategies i would keep it separated:

In DRAMSys and gem5 we have:
PDNA
PDNP
SREF

it is a necessary feature to keep it seperated.

**Comment by Sv3n** [comment]
(this thread is old, but still relevant, so here's a question):

I understand that we need an SREF command and a PDN command, however, I don't think we need distinct PDNA or PDNP commands. If a PDN command is given, while one or more banks are active, then this means the memory enters active powerdown mode. If all banks are precharged, then a PDN command means we enter precharged power-down mode. This is completely unambiguous, and closer to how a real DRAM behaves.

I propose to:

 * Add a flag variable to select either fast or slow mode. I'll have to see where it fits best (memspec, command line / class initialization, or somewhere else, or maybe everywhere).
 * For backwards compatibility, we can retain all the power down/up commands for now, but simply dispatch them all to 1 shared command handler. I'll add a PDN and PUP command.
 * Make our example use the new commands. Nothing should change in the output, but we will be able to simplify the command handlers.

**Comment by lsteiner-tukl** [comment]
In the latest standards there is no slow/fast entry any more.
